analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.CSharp8.cs

```diff
@@ -0,0 +1,209 @@
+﻿public class NullCoalescenceAssignment
+{
+    public void NullCoalescenceAssignment_NotNull(string s1)
+    {
+        s1 ??= "N/A";
+        s1.ToString(); // Compliant
+    }
+
+    public void NullCoalescenceAssignment_Null(string s1)
+    {
+        s1 ??= null;
+        s1.ToString(); // FIXME Non-compliant
```

Comment:
I'm not sure about non-compliancy here, S2259 should raise in this case (too?)

Reply:
Marked as FN. When `.ToString()` is reached we still don't know if `s1` is null or not, and since it's a method argument, I think it falls to S3900 instead of S2259.